heudiconv/convert.py

```diff
@@ -250,7 +250,7 @@ def update_complex_name(metadata, filename, suffix):
         Scan metadata dictionary from BIDS sidecar file.
     filename : str
         Incoming filename
-    suffix : str
+    suffix_counter : str
```

Comment:
well, `suffix_counter` is also confusing to me... we need to fix something up here:
- `counter` suggests an `int` of some kind but we have it as `str`
- its docstring says "An index ..." but the rest of it is not really providing much clarity to me
- we use that `suffix` (or `suffix_counter`) as a `%s` to be added to `_part-`
- `part` takes a `label` [bids spec](https://bids-specification.readthedocs.io/en/stable/99-appendices/09-entities.html#part) from a specific set of values (phase, mag, real and imag) not an index/counter.

so, overall -- we can't insert an index/count here as we would with counter... or what am I missing?

Reply:
I think this is a holdover from when heudiconv used `rec` for this information. Before #424, the "suffix" was just a number that could be used with `rec`:

https://github.com/nipy/heudiconv/blob/fdea77ce6263d3f92e51dd8a24ac506e7a2fce4e/heudiconv/convert.py#L512

https://github.com/nipy/heudiconv/blob/fdea77ce6263d3f92e51dd8a24ac506e7a2fce4e/heudiconv/convert.py#L553-L560

Perhaps the most BIDS-y solution would be to try to assign the right value to `part`, but, failing that, assign "suffix" (with a better variable name) to `rec`. So instead of `rec-magnitude`/`res-phase`/`rec-2` (the old way) or `part-mag`/`part-phase`/`part-2` (the current way), we would have `part-mag`/`part-phase`/`rec-2`, which would be BIDS-compliant.

I'm still not sure what to call `suffix`/`suffix_counter` though. It _is_ an index into the list of filenames, but that integer is converted to a string for the sake of adding it to the updated filename.

Reply:
let's just raise the RuntimeError on this `else:` and completely drop that suffix* kwarg.  The rationale is that we invoke this function only if `is_complex` (see https://github.com/nipy/heudiconv/pull/461/files#diff-d19d089741b3844c199c8236ec1c0fcd43400b2ce53cd587d203255add863c39R797) which is
```python
is_complex = 'M' in image_types and 'P' in image_types  # Determine if data are complex (magnitude + phase)
```
so we must have both M and P, and thus should not hit this `else`.  We are test covering this line of code just because we have a dedicated test for that function: https://github.com/tsalo/heudiconv/blob/master/heudiconv/tests/test_convert.py#L18

or am I missing smth?

Reply:
Oh, that's a good point! That seems like a great solution.

Reply:
Should be done in [e3e3d4f](https://github.com/nipy/heudiconv/pull/461/commits/e3e3d4f5b3a1a45ca886c9bcd6e0c6f3775ce256). Let me know if that's not what you had in mind.